File: src/main/java/yuu/rpg/SignJob.java
package yuu.rpg;

import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.block.Sign;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.player.PlayerInteractEvent;

import java.util.UUID;


public class SignJob implements Listener {

    private RPG plugin;
    CustomConfig uuid;
    UUID id;
    Player p;
    FileConfiguration config;
    String Job;

    SignJob(RPG pl) {
        plugin = pl;
        plugin.getServer().getPluginManager().registerEvents(this, plugin);
    }

    @EventHandler(priority = EventPriority.HIGHEST)
        public void onPlayerInteractEvent(PlayerInteractEvent e) {

        if (e.getAction() != Action.RIGHT_CLICK_BLOCK) return;
        Block clickedBlock = e.getClickedBlock();
        Material material = clickedBlock.getType();
        if (material == Material.SIGN || material == Material.WALL_SIGN) {
            Sign sign = (Sign) clickedBlock.getState();
            String line1 = sign.getLine(0);
            uuid = new CustomConfig(plugin, "UUID.yml");
            config = uuid.getConfig();
            p = e.getPlayer();
            id = p.getUniqueId();
            Job = config.getString("UUID." + id + ".Job");

            JobLvSet();
            JobChange(line1);
        }
    }

    private void JobLvSet(){
        ymlcheck(Job);
        config.set("UUID." + id + ".Lv." + Job + ".Lv", p.getLevel());
        config.set("UUID." + id + ".Lv."+ Job + ".Exp", p.getExp());
        uuid.saveConfig();
    }

    private void JobChange(String str) {



        switch (str) {

            case "戦士":

                p.setExp((float) config.getDouble("UUID." + id + ".Lv.Warrior.Exp"));
                p.setLevel(config.getInt("UUID." + id + ".Lv.Warrior.Lv"));

                config.set("UUID." + id + ".Job", "Warrior");
                p.sendMessage("戦士になりました");
                break;

            case "魔術師":

                p.setExp((float) config.getDouble("UUID." + id + ".Lv.Mage.Exp"));
                p.setLevel(config.getInt("UUID." + id + ".Lv.Mage.Lv"));

                config.set("UUID." + id + ".Job", "Mage");
                p.sendMessage("魔術師になりました");
                break;

            case "狩人":

                p.setExp((float) config.getDouble("UUID." + id + ".Lv.Hunter.Exp"));
                p.setLevel(config.getInt("UUID." + id + ".Lv.Hunter.Lv"));

                config.set("UUID." + id + ".Job", "Hunter");
                p.sendMessage("狩人になりました");
                break;

            case "村人":

                p.setExp((float) config.getDouble("UUID." + id + ".Lv.Villager.Exp"));
                p.setLevel(config.getInt("UUID." + id + ".Lv.Villager.Lv"));

                config.set("UUID." + id + ".Job", "Villager");
                p.sendMessage("村人になりました");
                break;

            case "放浪者":

                p.setExp((float) config.getDouble("UUID." + id + ".Lv.Wanderer.Exp"));
                p.setLevel(config.getInt("UUID." + id + ".Lv.Wanderer.Lv"));

                config.set("UUID." + id + ".Job", "Wanderer");
                p.sendMessage("放浪者になりました");
                break;

            case "騎士":

                p.setExp((float) config.getDouble("UUID." + id + ".Lv.Knight.Exp"));
                p.setLevel(config.getInt("UUID." + id + ".Lv.Knight.Lv"));

                config.set("UUID." + id + ".Job", "Knight");
                p.sendMessage("騎士になりました");
                break;

            case "クラフター":

                p.setExp((float) config.getDouble("UUID." + id + ".Lv.Crafter.Exp"));
                p.setLevel(config.getInt("UUID." + id + ".Lv.Crafter.Lv"));

                config.set("UUID." + id + ".Job", "Crafter");
                p.sendMessage("クラフターになりました");
                break;

        }
        uuid.saveConfig();
    }

    private void ymlcheck(String str) {

        String Lv = config.getString("UUID." + id + ".Lv." + str + ".Lv");
        String Exp = config.getString("UUID." + id + ".Lv." + str + ".Exp");

        if (Lv == null)
            config.set("UUID." + id + ".Lv." + str + ".Lv", 0);

        if (Exp == null)
            config.set("UUID." + id + ".Lv." + str + ".Exp", 0);
    }
}

File: src/main/java/yuu/rpg/SignJob.kt
package yuu.rpg

import org.bukkit.Material
import org.bukkit.block.Block
import org.bukkit.block.Sign
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.block.Action
import org.bukkit.event.player.PlayerInteractEvent

import java.util.UUID


class SignJob internal constructor(private val plugin: RPG) : Listener {

    init {
        plugin.server.pluginManager.registerEvents(this, plugin)
    }

    private var id: UUID? = null
    private var job: String? = null
    private val uuid: CustomConfig = CustomConfig(plugin, "UUID.yml")
    private val config: FileConfiguration? = uuid.getConfig()

    @EventHandler(priority = EventPriority.HIGHEST)
    fun onPlayerInteractEvent(e: PlayerInteractEvent) {

        if (e.action != Action.RIGHT_CLICK_BLOCK) return
        val clickedBlock = e.clickedBlock
        val material = clickedBlock.type
        if (material == Material.SIGN || material == Material.WALL_SIGN) {
            val sign = clickedBlock.state as Sign
            val line1 = sign.getLine(0)
            val p: Player = e.player
            id = p.uniqueId
            job = config!!.getString("UUID.$id.Job")

            jobchange(line1, p)
        }
    }

    private fun jobchange(str: String, p: Player) {

        ymlcheck(job)
        config!!.set("UUID.$id.Lv.$job.Lv", p.level)
        config.set("UUID.$id.Lv.$job.Exp", p.exp)

        val job2: String = JobUtil.jobjptoen(str)
        if (!job2.isEmpty()) {
            p.level = config.getInt("UUID.$id.Lv.$job2.Lv")
            p.exp = config.getDouble("UUID.$id.Lv.$job2.Exp").toFloat()

            config.set("UUID.$id.Job", job2)
            p.sendMessage(str + "になりました")
        }
        uuid.saveConfig()
    }

    private fun ymlcheck(str: String?) {

        if (config!!.getString("UUID.$id.Lv.$str.Lv") == null)
            config.set("UUID.$id.Lv.$str.Lv", 0)

        if (config.getString("UUID.$id.Lv.$str.Exp") == null)
            config.set("UUID.$id.Lv.$str.Exp", 0)
    }
}